src/main/java/com/example/springbootbookshop/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,52 @@
 package com.example.springbootbookshop.service.impl;
 
+import com.example.springbootbookshop.dto.BookDto;
+import com.example.springbootbookshop.dto.CreateBookRequestDto;
 import com.example.springbootbookshop.entity.Book;
+import com.example.springbootbookshop.mapper.BookMapper;
 import com.example.springbootbookshop.repository.BookRepository;
 import com.example.springbootbookshop.service.BookService;
+import jakarta.persistence.EntityNotFoundException;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 
+@RequiredArgsConstructor
 @Component
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public Book save(CreateBookRequestDto book) {
+        return bookRepository.save(bookMapper.toBook(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException("Can`t find"
+                        + "book with id:" + id));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto update(CreateBookRequestDto bookRequestDto, Long id) {
+        Book book = bookMapper.toBook(bookRequestDto);
+        book.setId(id);
+        bookRepository.save(book);
```

Comment:
it makes sense to first fetch book by id (otherwise you'll be able to update unexisting book)

then you can leverage mapstruct in order to update the book. it will be something like:
```
@Mapping(...) // add if necessary
void updateBook(CreateBookRequestDto dto, @MappingTarget Book book)
```